internal/persistence/postgres/options.go

```diff
@@ -0,0 +1,108 @@
+package postgres
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/kong/koko/internal/persistence"
+	"github.com/samber/lo"
+	"go.uber.org/zap"
+)
+
+var (
+	defaultSSLMode            = "verify-full"
+	unsupportedParamErrPrefix = "the '%s' parameter is unsupported, "
+	unsupportedParamErrs      = map[string]string{
+		"host":        unsupportedParamErrPrefix + " please instead use 'postgres.hostname'",
+		"port":        unsupportedParamErrPrefix + " please instead use 'postgres.port'",
+		"dbname":      unsupportedParamErrPrefix + " please instead use 'postgres.db_name'",
+		"user":        unsupportedParamErrPrefix + " please instead use 'postgres.user'",
+		"password":    unsupportedParamErrPrefix + " please instead use 'postgres.password'",
+		"sslrootcert": unsupportedParamErrPrefix + " please instead use 'postgres.ca_bundle_path'",
+	}
+)
+
+type Opts struct {
+	DBName           string
+	Hostname         string
+	ReadOnlyHostname string
+	Port             int
+	User             string
+	Password         string
+	EnableTLS        bool
+	CABundleFSPath   string
+	SQLOpen          persistence.SQLOpenFunc
```

Comment:
Please make sure to add docs for this external options structure.

Reply:
done